Class/C-Array-Methods/exercise1.js

```diff
@@ -17,7 +17,13 @@ let newArray = [];
 // Call the isTrainee function for each value
 // add each name to the new array only if the name is NOT a mentor
 
+for (let i = 0; i < namesList.length; i++) {
+  let name = isTrainee(namesList[i]); // not needed but added to help readability
 
+  if (name) {
+    newArray.push(namesList[i]);
+  }
+}

```

Comment:
This works but it is a bit of a complicated way of doing it.

You could simplify it to:

```js
if (isTrainee(namesList[i])) {
    newArray.push(namesList[i]);
}
```

This is also better from the point of view of understanding the code - since `name` looks like it ought to hold a string value such as `namesList[i]` rather than a boolean. Clear variable names become very important as code gets more complicated to ensure that you or other people looking at the code can understand properly what is going on.